src/main/java/frc/robot/commands/LiftClimbAndPull.java

```diff
@@ -9,82 +10,121 @@
+	};
+
+	@Override
+	public void initialize() {
+		// m_lift.moveToHeight(Constants.LiftConstants.StartClimbHeight);
+		List<Pose2d> Stages = Arrays.asList(new Pose2d[]{
+			FieldPose2024.StageSource.getCurrentAlliancePose(),
+			FieldPose2024.StageAmp.getCurrentAlliancePose(),
+			FieldPose2024.StageFar.getCurrentAlliancePose()});
+		Pose2d closestChain = m_swerveDrive.getPose().nearest(Stages);
+		double x = DISTANCE_FROM_TAG * Math.cos(closestChain.getRotation().getRadians());
+		double y = DISTANCE_FROM_TAG * Math.sin(closestChain.getRotation().getRadians());
+		Translation2d pose = closestChain.getTranslation().plus(new Translation2d (x, y));
+		m_startClimbPose = new Pose2d(pose, closestChain.getRotation());
+		m_swerveDrive.setTarget(new Pose2d(pose, closestChain.getRotation()));
+	}
+
+	@Override
+	public void execute() {
+		if (state == LiftSequence.MoveToArea) {
+			// Swerve odometry based movement
+
+			if(m_swerveDrive.atTarget(/* 1.5 */) && m_lift.atTargetHeight(LiftConstants.StartClimbHeight)) {
+				state = LiftSequence.VisionMoveAdjust;
+				m_swerveDrive.stop();
+			} else {
+				m_swerveDrive.moveToTarget(0.40);
+				m_lift.moveToHeight(LiftConstants.StartClimbHeight);
+			}
+
+		} else if ( state == LiftSequence.VisionMoveAdjust) {
+			// Vision based movement
+
+			Pose2d visionPose = m_vision.getPose();
+			Translation2d translationErrorMeters = visionPose.getTranslation().minus(m_startClimbPose.getTranslation());
+			Rotation2d rotationError = visionPose.getRotation().minus(m_startClimbPose.getRotation());
+			m_lift.moveToHeight(LiftConstants.StartClimbHeight);
+
+			if(translationErrorMeters.getNorm() < VISION_TRANSLATION_TOLERANCE
+					&& rotationError.getDegrees() < VISION_ROTATION_TOLERANCE
+					&& m_lift.atTargetHeight(LiftConstants.StartClimbHeight))
+			{
+				// Once we are close enough to the target orientation and the list is raised all the way, we can move into the next state
+				state = LiftSequence.Backup;
+
+				double x = -(DISTANCE_FROM_TAG-Constants.robotLengthMeters) * Math.cos(m_swerveDrive.getPose().getRotation().getRadians());
+				double y = -(DISTANCE_FROM_TAG-Constants.robotLengthMeters) * Math.sin(m_swerveDrive.getPose().getRotation().getRadians());
+				Translation2d currntPose = m_swerveDrive.getPose().getTranslation();
+				currntPose = currntPose.plus(new Translation2d(x, y));
+
+				m_swerveDrive.setTarget(new Pose2d(currntPose, m_swerveDrive.getPose().getRotation()));
+				
+			} else { 
+				var robotPose = m_swerveDrive.getPose();
+				m_swerveDrive.setTarget(new Pose2d(translationErrorMeters.plus(robotPose.getTranslation()), rotationError.plus(robotPose.getRotation())));
+				m_swerveDrive.moveToTarget(VISION_MOVE_SPEED);
+			}
+
+		} else if(state == LiftSequence.Backup) {
+			// Lifting the launcher up the elevator
+
+			if (m_swerveDrive.atTarget()) {
+				// Once the drive is at the target, we can move the lift all the way back down.
+				state = LiftSequence.PullDown;
```

Comment:
Some cases we need to handle - when raising the lift, we need to make sure the tilt is not too high. 
When pulling down, we need to make the tilt is above our min climb tilt angle (I think 16 degrees). 
And does that logic belong here? Or in the tilt/launcher?

This is logic we can maybe split up tomorrow?

Reply:
Is that to keep the tilt of the launcher at better angle for once it's up against the wood? To mitigate any resistance while trying to aim for the trap? Or is this a tolerance concern that the wrong angle might push us a little farther from the desired location? The purpose of this was just to get the robot up onto the chain in the event we can't see the chain well (but we do have the states to also attempt to score the trap as well if we want to do that later on).

In either case, I would do that work in LiftSequence.VisionMoveAdjust before the robot is pressed against the wall. It should be very similar to the code to raise the lift. We should only need 2-3 lines to fit it there and the launcher is already in the command.

Reply:
The main concern is that if the tilt is too low, the large gear teeth can run into the chain I think

And the max angle is to make sure we don't extend past the max robot height

And yeah, I think if we set the angle to the max min climb angle at the beginning, it covers both cases.